draft-ietf-oauth-sd-jwt-vc.md

```diff
@@ -328,10 +328,7 @@ verification key for the Issuer-signed JWT corresponds to the `iss` value:
 
 - JWT VC Issuer Metadata: If a recipient supports JWT VC Issuer Metadata and if the `iss` value contains an HTTPS URI, the recipient MUST
 obtain the public key using JWT VC Issuer Metadata as defined in (#jwt-vc-issuer-metadata).
-- X.509 Certificates: If the recipient supports X.509 Certificates, the recipient MUST obtain the public key from the leaf X.509 certificate defined by the `x5c` JWT header parameters of the Issuer-signed JWT and validate the X.509
-certificate chain in the following cases:
-    - If the `iss` value contains a DNS name encoded as a URI using the DNS URI scheme [@RFC4501], the DNS name MUST match a `dNSName` Subject Alternative Name (SAN) [@RFC5280] entry of the leaf certificate.
-    - In all other cases, the `iss` value MUST match a `uniformResourceIdentifier` SAN entry of the leaf certificate.
+- X.509 Certificates: If the recipient supports X.509 Certificates, the recipient MUST obtain the public key from the end-entity certificate of the certificates from the `x5c` header parameter of the Issuer-signed JWT and validate X.509 certificate chain accordingly. It MUST also ensure that the `iss` value matches a `uniformResourceIdentifier` SAN entry of the end-entity certificate or that the domain name in the `iss` value matches the `dNSName` SAN entry of the end-entity certificate.
```

Comment:
I find the following part confusing

> or that the domain name in the `iss` value

I believe it would be more clear something like:

It MUST also ensure that the `iss` value matches a `uniformResourceIdentifier` or a `dNSName` SAN entry of the end-entity certificate.

Reply:
PS: After this PR, my understanding is that the following options would be available

- `iss` is HTTPS URL, `x5c` is not provided, `kid` maybe used (recommended) => Resolve pub key via metadata
- `iss` is DID, `x5c is not provided, `kid` is DID-URL => Resolve pub key via DID resolution
- `iss` is a HTTPS URL, `x5c` is provided  => Pub Key is the leaf certificate, `iss` must be a SAN `uniformResourceIdentifier` entry
- `iss` is a string, `x5c` is provided => Pub key is the leaf certificate, `iss` must be a SAN `dNSName`  entry

Reply:
Sigh, yeah. I was just going to say something about iss always having to be an HTTPS URL. And then I realized I was confused about what layer what restriction is coming from due to being careless and having carelessly internalized some of the discussion in https://github.com/oauth-wg/oauth-sd-jwt-vc/issues/232 and https://github.com/openid/oid4vc-haip-sd-jwt-vc/issues/103.

Reply:
```suggestion
- X.509 Certificates: If the recipient supports X.509 Certificates and the `iss` value contains an HTTPS URI, the recipient MUST obtain the public key from the end-entity certificate of the certificates from the `x5c` header parameter of the Issuer-signed JWT and validate X.509 certificate chain accordingly. It MUST also ensure that the `iss` value matches a `uniformResourceIdentifier` SAN entry of the end-entity certificate or that the domain name in the `iss` value matches the `dNSName` SAN entry of the end-entity certificate.
```

Reply:
Dear @bc-pi 
Thanks for the clarification. I think it is clear now.

Reply:
Thanks for catching the mistake